app/src/main/java/com/example/androidpracticumcustomview/ui/theme/CustomContainerCompose.kt

```diff
@@ -1,35 +1,128 @@
 package com.example.androidpracticumcustomview.ui.theme
 
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.tween
 import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 
+private object CustomContainerConstants {
+    const val SIZE_CHECK_DELAY_MS = 50L
+}
 
-/*
-Задание:
-Реализуйте необходимые компоненты;
-Создайте проверку что дочерних элементов не более 2-х;
-Предусмотрите обработку ошибок рендера дочерних элементов.
-Задание по желанию:
-Предусмотрите параметризацию длительности анимации.
- */
 @Composable
 fun CustomContainerCompose(
-    firstChild: @Composable (() -> Unit)?,
-    secondChild: @Composable (() -> Unit)?
+    firstChild: @Composable (() -> Unit)? = null,
+    secondChild: @Composable (() -> Unit)? = null,
+    alphaAnimationDurationMs: Int = 2000,
+    translationAnimationDurationMs: Int = 5000,
 ) {
-    // Блок создания и инициализации переменных
-    // ..
+    val parentHeightPx = remember { mutableFloatStateOf(0f) }
+    val firstHeightPx = remember { mutableFloatStateOf(0f) }
+    val secondHeightPx = remember { mutableFloatStateOf(0f) }
+    val gapPx = with(LocalDensity.current) { 10.dp.toPx() }
 
-    // Блок активации анимации при первом запуске
-    LaunchedEffect(Unit) {
-        // TODO
-        // ...
-    }
+    val firstAlpha = remember { Animatable(0f) }
+    val secondAlpha = remember { Animatable(0f) }
+    val firstTranslateYPx = remember { Animatable(0f) }
+    val secondTranslateYPx = remember { Animatable(0f) }
+
+    val firstStarted = remember { mutableStateOf(false) }
+    val secondStarted = remember { mutableStateOf(false) }
```

Comment:
Здесь можно делегатом воспользоваться, чтобы каждый раз не обращаться к `value`:
```suggestion
    var firstStarted by remember { mutableStateOf(false) }
    var secondStarted by remember { mutableStateOf(false) }
```

Reply:
@ArturNurtdinov Спасибо!